File: core/applib/src/main/java/org/apache/isis/applib/layout/v1_0/TabGroup.java
/*
 *  Licensed to the Apache Software Foundation (ASF) under one
 *  or more contributor license agreements.  See the NOTICE file
 *  distributed with this work for additional information
 *  regarding copyright ownership.  The ASF licenses this file
 *  to you under the Apache License, Version 2.0 (the
 *  "License"); you may not use this file except in compliance
 *  with the License.  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing,
 *  software distributed under the License is distributed on an
 *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 *  KIND, either express or implied.  See the License for the
 *  specific language governing permissions and limitations
 *  under the License.
 */
package org.apache.isis.applib.layout.v1_0;

import java.io.Serializable;
import java.util.ArrayList;
import java.util.List;

import javax.xml.bind.annotation.XmlElement;
import javax.xml.bind.annotation.XmlTransient;
import javax.xml.bind.annotation.XmlType;

@XmlType()
public class TabGroup implements Serializable {

    private static final long serialVersionUID = 1L;

    // must be at least one tab.
    private List<Tab> tabs = new ArrayList<Tab>(){{
        add(new Tab());
    }};



    // no wrapper
    @XmlElement(name = "tab", required = true)
    public List<Tab> getTabs() {
        return tabs;
    }

    public void setTabs(List<Tab> tabs) {
        this.tabs = tabs;
    }



    private DomainObject owner;

    /**
     * Owner.
     *
     * <p>
     *     Set programmatically by framework after reading in from XML.
     * </p>
     */
    @XmlTransient
    public DomainObject getOwner() {
        return owner;
    }

    public void setOwner(final DomainObject owner) {
        this.owner = owner;
    }

}

File: incubator/clients/kroviz/src/main/kotlin/org/apache/causeway/client/kroviz/to/bs3/TabGroup.kt
/*
 *  Licensed to the Apache Software Foundation (ASF) under one
 *  or more contributor license agreements.  See the NOTICE file
 *  distributed with this work for additional information
 *  regarding copyright ownership.  The ASF licenses this file
 *  to you under the Apache License, Version 2.0 (the
 *  "License"); you may not use this file except in compliance
 *  with the License.  You may obtain a copy of the License at
 *
 *        http://www.apache.org/licenses/LICENSE-2.0
 *
 *  Unless required by applicable law or agreed to in writing,
 *  software distributed under the License is distributed on an
 *  "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
 *  KIND, either express or implied.  See the License for the
 *  specific language governing permissions and limitations
 *  under the License.
 */
package org.apache.causeway.client.kroviz.to.bs3

import org.w3c.dom.Node
import org.w3c.dom.asList

class TabGroup(node: Node) {
    var tabList = mutableListOf<Tab>()
    lateinit var metadataError: String
    lateinit var cssClass: String

    init {
        val nodeList = node.childNodes.asList()

        val tnList = nodeList.filter { it.nodeName.equals("bs3:tab") }
        for (n: Node in tnList) {
            val tab =Tab(n)
            tabList.add(tab)
        }
    }

    fun getPropertyList(): List<Property> {
        val list = mutableListOf<Property>()
        tabList.forEach { t ->
            list.addAll(t.getPropertyList())
        }
        return list
    }

}
